cprover_bindings/src/goto_program/stmt.rs

```diff
@@ -193,16 +194,21 @@ impl Stmt {
     /// `assert(cond, property_class, commment);`
     pub fn assert(cond: Expr, property_name: &str, message: &str, loc: Location) -> Self {
         assert!(cond.typ().is_bool());
+        assert!(!property_name.is_empty() && !message.is_empty());
+
+        // Create a Property Location Variant from any given Location type
+        let loc_with_property =
+            Location::create_location_with_property(message, property_name, loc);
 
         // Chose InternedString to seperate out codegen from the cprover_bindings logic
         let property_class = property_name.to_string().intern();
         let msg = message.into();
 
-        stmt!(Assert { cond, property_class, msg }, loc)
+        stmt!(Assert { cond, property_class, msg }, loc_with_property)
```

Comment:
is this the root cause of the bug? seems like this went from accidentally ignoring them to using them?